students/nDruP/lesson04/mailroom2.py

```diff
@@ -0,0 +1,167 @@
+3. At any point, the user should be able to quit their current task and return
+to the original prompt
+4. From the original prompt, the user should be able to quit the scipt cleanly
+"""
+
+
+import sys
+import os
+import datetime
+
+
+donor_dict = {"Sleve McDichael": [86457.89, 2346.43, 9099.09],
+              "Willie Dustice": [505.05, 43.21],
+              "Rey McScriff": [666.00],
+              "Mike Truk": [70935.30, 12546.70, 312.00],
+              "Bobson Dugnutt": [1234.56, 789.00],
+              "Todd Bonzalez": [715867.83, 10352.07]}
+
+
+def main_menu():
+    """
+    Prompt user to send a Thank You, Create a report, create letters, or quit.
+    """
+    user_prompt = None
+    valid_prompts = {"1": craft_thank_u,
+                     "2": create_report,
+                     "3": create_letters,
+                     "4": sys.exit}
+    while True:
+        print_divider()
+        print("We're a Pyramid Scheme & So Are You! E-Mailroom")
+        print_divider()
+        while user_prompt not in valid_prompts:
+            print("Please choose from the following options (1,2,3,4): ")
+            print("1. Send a Thank you")
+            print("2. Create Donor Report")
+            print("3. Send letters to everyone")
+            print("4. Quit")
+            user_prompt = input(">")
+        print_divider()
+        valid_prompts.get(user_prompt)()
+        input("\nPress enter to continue...")
+        user_prompt = None
+
+
+def craft_thank_u():
+    """
+    Prompt for a full name.
+    Prompt->"list": show a list of the donor names
+    Prompt->name not in list: add name to list;
+    Then prompt for donation amount, add amount to donation history
+    Compose and print an email thanking the donor for their donation.
+    Return to main
+    """
+    donor_name = "list"
+    donation_amt = None
+    message = ''
+    while True:
+        print("Let's craft a very personal thank you note for our donor!")
+        print("Return to the main menu at any time by entering 'exit'")
+
+        print("Pull up a list of donor names by entering 'list'")
+        while donor_name == "list":
+            donor_name = input("Enter Donor Name >")
+            if donor_name.lower() == "list":
+                print(("{}\n"*len(donor_dict)).format(*donor_dict))
+        if donor_name.lower() == "exit":
+            break
+
+        donation_amt = input("Enter their Donation Amount >")
+        if donation_amt.lower() == "exit":
+            break
+        donation_amt = float(donation_amt)
+
+        if donor_name not in donor_dict:
+            donor_dict[donor_name] = []
+        donor_dict[donor_name].append(donation_amt)
+        print_divider()
+        message = f"Dearest {donor_name},\n"
+        message += f"Thank you so much for donation of ${donation_amt:.2f}!\n"
+        message += "We will use your donation to create a real living Pokemon."
+        message += f"\nSincerely,\nWe're a Pyramid Scheme & so is {donor_name}"
```

Comment:
note that there are two more ways to do multi-strings that are a bit more readable

use triple quotes (in this case no need for \n)

```
s = f"""Dearest {donor_name}
Thank you so much...
"""

```
or with parens (you will need \n)
```
>>> s = (f"Dearest {donor_name}\n"
...     "thank you so much...")
>>>
```